backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/modelvalidator/C_Order.java

```diff
@@ -34,12 +35,22 @@
 @Interceptor(I_C_Order.class)
 public class C_Order
 {
-	// @DocValidate(timings = { ModelValidator.TIMING_AFTER_COMPLETE, ModelValidator.TIMING_AFTER_REACTIVATE, ModelValidator.TIMING_AFTER_CLOSE })
-	// public void invalidateInvoiceCandidates(final I_C_Order order)
-	// {
-	// 	final IInvoiceCandidateHandlerBL invoiceCandidateHandlerBL = Services.get(IInvoiceCandidateHandlerBL.class);
-	// 	invoiceCandidateHandlerBL.invalidateCandidatesFor(order);
-	// }
+	private final IInvoiceCandidateHandlerBL invoiceCandidateHandlerBL = Services.get(IInvoiceCandidateHandlerBL.class);
+	private final IBPartnerStatsBL bpartnerStatsBL = Services.get(IBPartnerStatsBL.class);
+	private final IBPartnerStatsDAO bpartnerStatsDAO = Services.get(IBPartnerStatsDAO.class);
+	private final ICurrencyBL currencyBL = Services.get(ICurrencyBL.class);
+	private final IDocTypeDAO docTypeDAO = Services.get(IDocTypeDAO.class);
+	private final ISysConfigBL sysConfigBL = Services.get(ISysConfigBL.class);
+
+	@DocValidate(timings = {
+			ModelValidator.TIMING_AFTER_COMPLETE,
+			ModelValidator.TIMING_AFTER_REACTIVATE,
```

Comment:
imho only for ModelValidator.TIMING_AFTER_COMPLETE,
			ModelValidator.TIMING_AFTER_REACTIVATE
 is needed

Reply:
also can you pls add a short comment explaining why we need to invalidate it on COMPLETE.
something like
```
we have to invalidate on complete, for the case the order was re-activated before and now completed again and user changed things in the order (e.g. the product)
```